suite-common/wallet-utils/src/transactionUtils.ts

```diff
@@ -42,20 +43,16 @@ const generateTransactionDateKey = (d: Date) =>
 
 /** Parse Date object from a string in YYYY-MM-DD format */
 export const parseTransactionDateKey = (key: string) => {
-    const parts = key.split('-');
-    const d = new Date(Number(parts[0]), Number(parts[1]) - 1, Number(parts[2]));
-    return d;
+    const [year, month, day] = key.split('-');
+    return new Date(Number(year), Number(month) - 1, Number(day));
 };
 
-/* Convert date to string in YYYY-MM format */
-const generateTransactionMonthKey = (d: Date) => `${d.getFullYear()}-${d.getMonth() + 1}`;
+export type MonthKey = string & { __type: 'MonthKey' };
+const generateTransactionMonthKey = (d: Date): MonthKey =>
+    // Adding days because of time zones of UTC
+    addDays(startOfMonth(d), 1).toUTCString() as MonthKey;
```

Comment:
Or maybe just rename `generateTransactionMonthKey` to something like `generateTransactionDateTimeKey`. This naming is confusing.

Reply:
No, because we need to have it in UTC format. If it's not in UTC, there will still be bug which was there when months were shifted by one. 

I can rename it but it's more confusing then before, because now there will be two functions with very similar name which makes it unclear why we're doing it. I think it was better name before, because it was generating month keys for ordering, but here's a fixup commit and let's see what others think. 

https://github.com/trezor/trezor-suite/pull/7506/commits/f03a8634cf9db18c1edd7a932eaca80f518dbd36